pkg/jtracer/jtracer.go

```diff
@@ -14,16 +14,95 @@
 
 package jtracer
 
-import "github.com/opentracing/opentracing-go"
+import (
+	"context"
+	"fmt"
+	"sync"
+
+	"github.com/opentracing/opentracing-go"
+	"go.opentelemetry.io/otel"
+	otbridge "go.opentelemetry.io/otel/bridge/opentracing"
+	"go.opentelemetry.io/otel/exporters/otlp/otlptrace"
+	"go.opentelemetry.io/otel/exporters/otlp/otlptrace/otlptracegrpc"
+	"go.opentelemetry.io/otel/propagation"
+	"go.opentelemetry.io/otel/sdk/resource"
+	sdktrace "go.opentelemetry.io/otel/sdk/trace"
+	semconv "go.opentelemetry.io/otel/semconv/v1.4.0"
+	"go.opentelemetry.io/otel/trace"
+)
 
 type JTracer struct {
-	OT opentracing.Tracer
+	OT     opentracing.Tracer
+	OTEL   trace.TracerProvider
+	closer func() error
 }
 
-func OT(t opentracing.Tracer) JTracer {
-	return JTracer{OT: t}
+var once sync.Once
+
+func New() (*JTracer, error) {
+	jt := JTracer{}
+	opentracingTracer, otelTracerProvider, closed, error := jt.initBoth(context.Background())
+
+	return &JTracer{
+		OT:     opentracingTracer,
+		OTEL:   otelTracerProvider,
+		closer: closed,
+	}, error
```

Comment:
As these codcov warnings suggest, it would be nice to add some tests for this file.

Reply:
Yes, TODO!